apps/api/src/donations/donations.service.ts

```diff
@@ -114,6 +115,87 @@ export class DonationsService {
     return donation
   }
 
+  /**
+   * Create initial donation object for tracking purposes
+   * This is used when the payment is created from the payment intent
+   */
+  async createInitialDonationFromIntent(
+    campaign: Campaign,
+    stripePaymentDto: CreateStripePaymentDto,
+    paymentIntent: Stripe.PaymentIntent,
+  ): Promise<Donation> {
+    Logger.debug('[ CreateInitialDonationFromIntent]', {
+      campaignId: campaign.id,
+      amount: paymentIntent.amount,
+      paymentIntentId: paymentIntent.id,
+    })
+
+    /**
+     * Create or connect campaign vault
+     */
+    const vault = await this.prisma.vault.findFirst({ where: { campaignId: campaign.id } })
+    const targetVaultData = vault
+      ? // Connect the existing vault to this donation
+        { connect: { id: vault.id } }
+      : // Create new vault for the campaign
+        { create: { campaignId: campaign.id, currency: campaign.currency, name: campaign.title } }
+
+    /**
+     * Create or update initial donation object
+     */
+    const donation = await this.prisma.donation.upsert({
```

Comment:
This is an upsert call since I ran into `409 Conflict` errors, because when that method is called the `payment_intent.create` hook had already created a donation with that `extPaymentIntentId` and it should be unique for every donation.

Maybe it should be a straight `update` call since the frontend will call the `create-stripe-payment` always after the `payment_intent.created` has been sent